### PR TITLE
fix(chore): use a callable for defaults

### PR DESCRIPTION
**Pull request by lecrepont01**
Without a callable the defaults are evaluated on import which causes unexpected errors with other commands.

### CODE REVIEW

**Review by sileht — DISMISSED** [review]
Lazy loading them is a good idea. 

We cannot just use a lambda otherwise cli arguments will be ignored and we will always load the default.

**Comment by sileht** [comment]
> Lazy loading them is a good idea.
> 
> We cannot just use a lambda otherwise cli arguments will be ignored and we will always load the default.

I remake a couple of tests and it seems, the issue occurs only when `is_flag=True`

**Comment by sileht** [comment]
I reproduce the issue in a simplier app https://github.com/sileht/click-bug :

```
# without lambda:
$ poetry run python3 main.py work -v
True
$ poetry run python3 main.py work
False

# with lambda:
$ poetry run python3 main.py buggy
False
$ poetry run python3 main.py buggy -v
False

**Comment by lecrepont01** [comment]
> I reproduce the issue in a simplier app https://github.com/sileht/click-bug :
> 
> ```
> # without lambda:
> $ poetry run python3 main.py work
> False
> $ poetry run python3 main.py work -v
> True
> 
> # with lambda:
> $ poetry run python3 main.py buggy
> False
> $ poetry run python3 main.py buggy -v
> False
> ```

ThanksI am not sure I understand why user input does not always prevail when it is passed but I'll take a better look to find a fix.

File: mergify_cli/stack/cli.py [hunk 1]
```diff
@@ -76,7 +76,13 @@ async def edit() -> None:
...
     "-k",
     is_flag=True,
-    default=asyncio.run(utils.get_default_keep_pr_title_body()),
+    # NOTE(leo): lazy loading default with `is_flag=True` is buggy in Click
+    # because it always evaluates and takes precedence on the user provided value.
+    # Use a callback instead to load the default.
+    callback=utils.get_default_keep_pr_title_body_callback,
+    # The None default allows us to lazy load the default in the callback when
+    # needed, with None meaning that the user did not provide the flag.
+    default=None,
```
Comment (sileht) [inline mergify_cli/stack/cli.py:85]: I talked with click people and to make is_flag work with dynamic default, the syntax is:

```suggestion
    flag_value=True
    default=lambda: asyncio.run(utils.get_default_keep_pr_title_body())
```

Comment (sileht) [reply mergify_cli/stack/cli.py:85]: Click introspects `default=` to set `flag_value` and when the `default` is a callable it assumes it's `True` and set `flag_value` to False`

Comment (lecrepont01) [reply mergify_cli/stack/cli.py]: Indeed works fine on the example, it's leaner. I removed my workaround.

**Review by sileht — APPROVED** [review]

**Pull request merged**